Bottom: 76ee2dfe59bc03f2d0578f11a7a4407ffc848cbd
Top:    ccc96d90888319a2b133f0d84ebc99813032350f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-09-12 01:22:10 +0100

Debug / attempt to fix DRC crash


---

diff --git a/src/find.c b/src/find.c
index 2f6dd76..dcce41c 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2389,6 +2389,10 @@ IsLineInPolygon (LineType *Line, PolygonType *Polygon)
        Coord wid = (Line->Thickness + Bloat + 1) / 2;
        Coord x1, x2, y1, y2;
 
+       printf ("Line thickness is %i, Bloat is %i\n",
+               Line->Thickness, Bloat);
+       printf ("Testing with wid=%i\n", wid);
+
        x1 = MIN (Line->Point1.X, Line->Point2.X) - wid;
        y1 = MIN (Line->Point1.Y, Line->Point2.Y) - wid;
        x2 = MAX (Line->Point1.X, Line->Point2.X) + wid;
diff --git a/src/polygon.c b/src/polygon.c
index 1dfb1bc..48aac13 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -1717,6 +1717,7 @@ bool
 IsRectangleInPolygon (Coord X1, Coord Y1, Coord X2, Coord Y2, PolygonType *p)
 {
   POLYAREA *s;
+  printf ("X1=%i, X2=%i, Y1=%i, Y2=%i\n", X1, X2, Y1, Y2);
   if (!
       (s = RectPoly (min (X1, X2), max (X1, X2), min (Y1, Y2), max (Y1, Y2))))
     return false;
